src/main/java/com/premiumminds/persistence/MultiplePersistenceTransaction.java

```diff
@@ -59,8 +59,20 @@ public void start() {
 	public void end() {
 		try {
 			boolean rollback = isRollbackOnly();
+			Exception caughtException = null;
 			for(PersistenceTransaction pt : persistenceTransactions){
-				pt.end();
+				try {
+					if (caughtException!=null) {
+						pt.setRollbackOnly();
+						rollback = true;
+					}
+				} catch (Exception e) { }
+				try {
+					pt.end();
+				} catch (Exception e) {
+					log.warn("Failed to end transaction at "+pt.getClass().getSimpleName()+": "+e.getMessage());
```

Comment:
You should send the exception to the logger and not just the message
log.warn(msg, exception)